### PR TITLE
Technical improvements to perlxstut

### PR DESCRIPTION
**Pull request by ghost**
For https://github.com/Perl/perl5/issues/18163.

### CODE REVIEW

File: dist/ExtUtils-ParseXS/lib/perlxstut.pod [hunk 1]
```diff
@@ -122,7 +122,7 @@ The file Makefile.PL should look something like this:
 	VERSION_FROM => 'Mytest.pm', # finds $VERSION
 	LIBS         => [''],   # e.g., '-lm'
 	DEFINE       => '',     # e.g., '-DHAVE_SOMETHING'
-	INC          => '',     # e.g., '-I/usr/include/other'
+	INC          => '-I.', # e.g., '-I. -I/usr/include/other'
```
Comment (Grinnz) [inline dist/ExtUtils-ParseXS/lib/perlxstut.pod]: By no means should `-I.` be recommended to users.

Comment (ghost) [reply dist/ExtUtils-ParseXS/lib/perlxstut.pod]: It's in the `h2xs -A -n Mytest` output Makefile.PL.

**Comment by apparluk** [comment]
@atoomic Have you a position regarding https://github.com/Perl/perl5/pull/18205#discussion_r499306286?

File: dist/ExtUtils-ParseXS/lib/perlxstut.pod (hunk 1 above)
Comment (tonycoz) [reply dist/ExtUtils-ParseXS/lib/perlxstut.pod]: Did you test run h2xs in a directory with a `*.h` file in it?

h2xs helpfully adds the `-I.` if it finds any `*.h` or `*.hh` files and I don't think that's the case if you follow the steps in perlxsitut.

Comment (ghost) [reply dist/ExtUtils-ParseXS/lib/perlxstut.pod]: https://perldoc.perl.org/perlxstut

EXAMPLE 1

Run `"h2xs -A -n Mytest"`.

```
bash-4.3$ ls /tmp/*.h
ls: cannot access '/tmp/*.h': No such file or directory
bash-4.3$ ls /tmp/*.hh
ls: cannot access '/tmp/*.hh': No such file or directory
bash-4.3$ cd /tmp && h2xs -A -n Mytest
Defaulting to backwards compatibility with perl 5.22.2
If you intend this module to be compatible with earlier perl versions, please
specify a minimum perl version with the -b option.

Writing Mytest/ppport.h
Writing Mytest/lib/Mytest.pm
Writing Mytest/Mytest.xs
Writing Mytest/Makefile.PL
Writing Mytest/README
Writing Mytest/t/Mytest.t
Writing Mytest/Changes
Writing Mytest/MANIFEST
bash-4.3$ grep -nr other Mytest/Makefile.PL
16:    INC               => '-I.', # e.g., '-I. -I/usr/include/other'
bash-4.3$ rm -rf Mytest
bash-4.3$ 
```

**Comment by xsawyerx** [comment]
Can you please copy over the remaining changes from #18165, so we could close that one in favor of this one?

**Comment by apparluk** [comment]
> Can you please copy over the remaining changes from #18165, so we could close that one in favor of this one?

Yes.

**Comment by apparluk** [comment]
@xsawyerx

This just lists a few hyperlinks on the page. 

DESCRIPTION

https://metacpan.org/pod/perlxs (Error 404 - Page not found - metacpan.org)
https://metacpan.org/pod/perlapi (Error 404 - Page not found - metacpan.org)

EXAMPLE 1

https://metacpan.org/pod/perlxs#The-PROTOTYPES:-Keyword  (Error 404 - Page not found - metacpan.org)

EXAMPLE 3

https://metacpan.org/pod/perlxstypemap  (Error 404 - Page not found - metacpan.org)

SEE ALSO

https://metacpan.org/pod/perlxs (Error 404 - Page not found - metacpan.org)
https://metacpan.org/pod/perlapi (Error 404 - Page not found - metacpan.org)

**Comment by Grinnz** [comment]
These documentation pages are created on install so metacpan cannot display them. The links will work on https://perldoc.perl.org.

**Comment by apparluk** [comment]
@Grinnz 

https://github.com/varehus/perl5/blob/perlxstut-update/dist/ExtUtils-ParseXS/lib/perlxstut.pod

Any other changes?

**Comment by apparluk** [comment]
Hold off on a merge.

AUTHORS was changed in another pull request.

After that other merge I'll revert the herein AUTHORS change.

**Comment by ghost** [comment]
@xsawyerx use this [branch](https://github.com/apparluk/perl5/tree/perlxstut-update) instead.

https://github.com/Perl/perl5/pull/18240

**Pull request closed without merging**